lib/https.js

```diff
@@ -34,6 +34,12 @@ const { urlToOptions, searchParamsSymbol } = require('internal/url');
 function Server(opts, requestListener) {
   if (!(this instanceof Server)) return new Server(opts, requestListener);
 
+  if (typeof opts === 'function') {
+    requestListener = opts;
+    opts = undefined;
+  }
+  opts = opts ? util._extend({}, opts) : {};
```

Comment:
@addaleax I was about to say that but I'm not sure it matters in the case of starting a server. \*shrug\*